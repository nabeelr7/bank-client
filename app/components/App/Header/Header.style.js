import styled from 'styled-components';
import { colors, media } from 'utils';
import {
  MenuUnfoldOutlined,
  MenuFoldOutlined,
  BarsOutlined,
} from '@ant-design/icons';
import { Layout } from 'antd';

export const headerHeight = '64px';

export const StyledHeader = styled(Layout.Header)`
  display: flex;
  align-items: center;
  padding: 0 25px;
  justify-content: space-between;
  width: 100%;
  position: fixed;
  height: ${headerHeight};
`;

export const StyledMenuUnfoldOutlined = styled(MenuUnfoldOutlined)`
  color: ${colors.white};
  font-size: 19px;
  display: none;

  ${media.tablet`
    display: contents;
  `}
`;

export const StyledMenuFoldOutlined = styled(MenuFoldOutlined)`
  color: ${colors.white};
  font-size: 19px;
  display: none;

  ${media.tablet`
    display: contents;
  `}
`;

export const StyledBarsOutlined = styled(BarsOutlined)`
  color: ${colors.white};
  font-size: 19px;
  display: contents;

  ${media.tablet`
    display: none;
  `}
`;
