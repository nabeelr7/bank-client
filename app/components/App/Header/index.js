/**
 *
 * Header
 *
 */

import React from 'react';

import { createStructuredSelector } from 'reselect';
import { useSelector, useDispatch } from 'react-redux';
import { makeSelectIsCollapsedSidebar } from 'containers/App/selectors';
import {
  collapsedSidebarAction,
  collapsedDrawerAction,
} from 'containers/App/actions';
import HeaderAction from 'components/App/HeaderAction';
import {
  StyledMenuUnfoldOutlined,
  StyledMenuFoldOutlined,
  StyledBarsOutlined,
  StyledHeader,
} from './Header.style';
import Mark from '../Mark';

const stateSelector = createStructuredSelector({
  isCollapsedSidebar: makeSelectIsCollapsedSidebar(),
});

export default function Header() {
  const { isCollapsedSidebar } = useSelector(stateSelector);
  const dispatch = useDispatch();
  const onCollapsedSidebar = () => dispatch(collapsedSidebarAction());
  const onCollapsedDrawer = () => dispatch(collapsedDrawerAction());

  return (
    <StyledHeader open={isCollapsedSidebar}>
      {isCollapsedSidebar ? (
        <StyledMenuUnfoldOutlined onClick={onCollapsedSidebar} />
      ) : (
        <StyledMenuFoldOutlined onClick={onCollapsedSidebar} />
      )}

      <StyledBarsOutlined onClick={onCollapsedDrawer} />

      <HeaderAction />

      <Mark />
    </StyledHeader>
  );
}
