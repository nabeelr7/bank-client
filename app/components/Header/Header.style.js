import styled from 'styled-components';

const StyledHeader = styled.header`
  display: flex;
  justify-content: space-between;
  padding: 0 40px;
  border-bottom: 1px solid rgba(0, 0, 0, 0.12);
`;

export default StyledHeader;
