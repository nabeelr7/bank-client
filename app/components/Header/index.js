/**
 *
 * Header
 *
 */

import React from 'react';
import { Link } from 'react-router-dom';
import icon from 'images/icon.png';
import logo from 'images/logo.png';
import { StyledHeader, StyledImg } from './Header.style';

// import PropTypes from 'prop-types';
// import styled from 'styled-components';

function Header() {
  return (
    <StyledHeader>
      <Link to="/">
        <StyledImg
          width="180"
          height="100%"
          src={logo}
          alt="Bank Application"
        />
      </Link>

      <StyledImg width="50" height="100%" src={icon} alt="Adrian Pietrzak" />
    </StyledHeader>
  );
}

Header.propTypes = {};

export default Header;
