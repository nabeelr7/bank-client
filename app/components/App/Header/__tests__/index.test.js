import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import configureStore from 'configureStore';
import Header from '../index';

describe('<Header />', () => {
  const history = createMemoryHistory();
  const store = configureStore({}, history);

  it('should match the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <Header />
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
