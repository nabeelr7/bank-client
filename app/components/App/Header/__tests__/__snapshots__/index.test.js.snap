// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match the snapshot 1`] = `
<header
  class="ant-layout-header Headerstyle__StyledHeader-en79cw-0 kWRNOn"
>
  <span
    aria-label="menu-fold"
    class="anticon anticon-menu-fold Headerstyle__StyledMenuFoldOutlined-en79cw-2 jqgddL"
    role="img"
    tabindex="-1"
  >
    <svg
      aria-hidden="true"
      class=""
      data-icon="menu-fold"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 000 13.8z"
      />
    </svg>
  </span>
  <span
    aria-label="bars"
    class="anticon anticon-bars Headerstyle__StyledBarsOutlined-en79cw-3 bJgLVe"
    role="img"
    tabindex="-1"
  >
    <svg
      aria-hidden="true"
      class=""
      data-icon="bars"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="0 0 1024 1024"
      width="1em"
    >
      <path
        d="M912 192H328c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h584c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 284H328c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h584c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 284H328c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h584c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM104 228a56 56 0 10112 0 56 56 0 10-112 0zm0 284a56 56 0 10112 0 56 56 0 10-112 0zm0 284a56 56 0 10112 0 56 56 0 10-112 0z"
      />
    </svg>
  </span>
</header>
`;
